src/System.Buffers.Experimental/System/Buffers/Sequences/SequenceExtensions.cs

```diff
@@ -12,24 +12,38 @@ public static class Sequence
         public static ReadOnlySpan<byte> ToSpan(this ReadOnlySequence<byte> sequence)
         {
             SequencePosition position = sequence.Start;
-            ResizableArray<byte> array = new ResizableArray<byte>(1024);
+            int capacity = 0;
+            while (sequence.TryGet(ref position, out ReadOnlyMemory<byte> buffer))
+                capacity += buffer.Length;
+
+            position = sequence.Start;
+            var count = 0;
+            byte[] array = new byte[capacity];
```

Comment:
Instead of looping through all the ROSequence segments, could this just be `sequence.Length` which gives you the length of the sequence already?

https://docs.microsoft.com/en-us/dotnet/api/system.buffers.readonlysequence-1.length?view=netcore-3.1

We could even add a check up front that if `sequence.Length > int.Max`, throw invalid operation exception to avoid accidental integer overflows.

```suggestion
            int count = 0;
            byte[] array = new byte[(int)sequence.Length];
```